scripts/profiler/data_collector.py

```diff
@@ -20,23 +51,26 @@ def main():
     args = parser.parse_args()
 
     if args.log is not None:
-	    log_lvl_number = int(getattr(logging, args.log.upper(), None))
+        log_lvl_number = int(getattr(logging, args.log.upper(), None))
     else:
-	    log_lvl_number = logging.DEBUG
-
-    def sigint_handler(sig, frame):
-        end_ev.set()
+        log_lvl_number = logging.INFO
 
-    signal.signal(signal.SIGINT, sigint_handler)
-    end_ev = threading.Event()
+    # event is made to ensure that ModelCreator class is initialized before Rtt2Socket starts sending data
+    event = Event()
 
-    profiler = RttNordicProfilerHost(
-                event_filename=args.dataset_name + ".csv",
-                finish_event=end_ev,
-                event_types_filename=args.dataset_name + ".json",
-                log_lvl=log_lvl_number)
-    profiler.get_events_descriptions()
-    profiler.read_events_rtt(args.time)
+    try:
+        p1 = Process(target=model_creator, args=(event, args.dataset_name, log_lvl_number,), daemon=True)
+        p2 = Process(target=rtt2socket, args=(event, log_lvl_number,), daemon=True)
+        p1.start()
+        p2.start()
+        time.sleep(args.time)
```

Comment:
Consider providing timeout to processes instead of killing them

Reply:
As far as I know there is no such option in Python multiprocessing.